src/components/WeatherTable/WeatherTable.tsx

```diff
@@ -0,0 +1,84 @@
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import MUITableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import TableCell from 'components/TableCell';
+import { Forecast } from 'types/forecast';
+import Location from 'types/location';
+import findCountryNameByCode from 'utils/findCountryNameByCode';
+import HOURS from './hours';
+import * as classes from './styles';
+
+interface Props {
+  forecast: Forecast;
+  location: Location;
+  className?: string;
+}
+
+const TIME_PERIODS_AMOUNT = 8;
+
+export default function WeatherTable({ forecast, location, className }: Props) {
+  return (
+    <Box className={className}>
+      <p css={classes.tableTitle}>
+        5-Day Weather Forecast for&nbsp;
+        {location &&
+          `${findCountryNameByCode(location.country)}, ${location.name}`}
+      </p>
+      <TableContainer
+        component={Paper}
+        css={classes.tableContainer}>
+        <Table aria-label='5-Day Weather Forecast'>
+          <TableHead>
+            <TableRow>
+              <MUITableCell css={classes.tableHeadCell}>
+                Date/Hours
+              </MUITableCell>
+              {HOURS.map((hour) => (
+                <MUITableCell
+                  align='center'
+                  css={classes.tableHeadCell}
+                  key={hour}>
+                  {hour}
+                </MUITableCell>
+              ))}
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {forecast?.map(([date, weather], index) => {
+              const emptyCells = TIME_PERIODS_AMOUNT - weather.length;
```

Comment:
What `weather.length` mean?

Reply:
It indicates the quantity of available weather data for today. We need this information to determine how many cells we should skip.